agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -999,6 +1003,46 @@ func (dg *dockerGoClient) listContainers(ctx context.Context, all bool) ListCont
 	return ListContainersResponse{DockerIDs: containerIDs, Error: nil}
 }
 
+func (dg *dockerGoClient) ListImages(ctx context.Context, timeout time.Duration) ListImagesResponse {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	response := make(chan ListImagesResponse, 1)
+	go func() { response <- dg.listImages(ctx) }()
+	select {
+	case resp := <-response:
+		return resp
+	case <-ctx.Done():
+		err := ctx.Err()
+		if err == context.DeadlineExceeded {
+			return ListImagesResponse{Error: &DockerTimeoutError{timeout, "listing"}}
+		}
+		return ListImagesResponse{Error: &CannotListImagesError{err}}
+	}
+}
+
+func (dg *dockerGoClient) listImages(ctx context.Context) ListImagesResponse {
+	client, err := dg.sdkDockerClient()
+	if err != nil {
+		return ListImagesResponse{Error: err}
+	}
+	images, err := client.ImageList(ctx, types.ImageListOptions{
+		All: false,
```

Comment:
can you add comment on what All flag signifies? with false, it's not clear what we are ignoring. From docker API:
"All: Show all images (default hides intermediate images)"
So I'm wondering if this is even required.